hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -241,7 +241,7 @@ class TableStage(
       repartitionNoShuffle(partitioner.strictify)
   }
 
-  def repartitionNoShuffle(newPartitioner: RVDPartitioner, allowDuplication: Boolean = false): TableStage = {
+  def repartitionNoShuffle(newPartitioner: RVDPartitioner, allowDuplication: Boolean = false, allowPartitionFilter: Boolean = false): TableStage = {
```

Comment:
Can you explain the difference in behavior if `allowPartitionFilter` is set or not?

Reply:
Is it: if true, the returned partitioner drops partitions of `newPartitioner` that would be empty, otherwise the returned partitioner is `newPartitioner`?

Reply:
This allows the output to be partitioned to a subset of `newPartitioner`, with empty partitions dropped.

If `allowPartitionFilter` is `false`, the output will always be partitioned to `newPartitioner`, even with empty partitions.

Reply:
Got it, thanks. It might be clearer to call it `dropEmptyPartitions`.

Reply:
good change 👍

Reply:
did the rename.